### PR TITLE
feat(alerts): Allow configuration of filter tabs via predefinedFilters

### PR DESCRIPTION
**Pull request by TilmanHaupt**
## Pull Request Details

Adding predefinedFilters to pluginDefinition to generalize them for different purposes.

## Breaking Changes

none

## Issues Fixed

Part of #258

### CODE REVIEW

File: alerts/README.md [hunk 1]
```diff
@@ -43,6 +43,10 @@ This plugin extension contains [Prometheus Alertmanager](https://github.com/prom
 | `silenceExcludedLabels` | SilenceExcludedLabels are labels that are initially excluded by default when creating a silence. However, they can be added if necessary when utilizing the advanced options in the silence form.The labels must be an array of strings. Example: `["pod", "pod_name", "instance"]`                                                            | `[]`           |
 | `filterLabels`          | FilterLabels are the labels shown in the filter dropdown, enabling users to filter alerts based on specific criteria. The 'Status' label serves as a default filter, automatically computed from the alert status attribute and will be not overwritten. The labels must be an array of strings. Example: `["app", "cluster", "cluster_type"]` | `[]`           |
 
+| `silenceTemplates` | silenceTemplates are used in the Modal (Schedule Silence) to allow pre-defined silences to be used to scheduled Maintenance Windows. The format consists of a list of objects including description, editable_labels (array of strings specifying the labels that users can modify), fixed_labels (map containing fixed labels and their corresponding values), status, and title. | `[]` |
```
Comment (ArtieReus) [inline alerts/README.md:46]: - `silenceTemplates are used...`: please use also capital letters
- `Modal (Schedule Silence)`: please in lowercase
- `Maintenance Windows`: lowercase

File: alerts/README.md [hunk 2]
```diff
@@ -43,6 +43,10 @@ This plugin extension contains [Prometheus Alertmanager](https://github.com/prom
...
+
+| `predefinedFilters` | PredefinedFilters are applied to Filter Alerts via TabNavigation. They are loaded by default when the application is loaded. The format is a list of objects including name, displayname and matchers (containing keys corresponding value) | `[]` |
```
Comment (ArtieReus) [inline alerts/README.md:48]: - ` Filter Alerts via TabNavigation`: lowercase

or what do you think about this description for the beginning:
PredefinedFilters are filters applied through TabNavigation in the UI to differentiate between contexts such as Production, QA, and others.

File: alerts/plugindefinition.yaml [hunk 3]
```diff
@@ -37,6 +37,10 @@ spec:
       description: "SilenceTemplates are pre-defined silence templates that can be used to scheduled Maintenance Windows. The format consists of a list of objects including description, editable_labels (array of strings specifying the labels that users can modify), fixed_labels (map containing fixed labels and their corresponding values), status, and title."
       required: false
       type: list
+    - name: predefinedFilters
+      description: "PredefinedFilters are filters that are applied by default when the application is loaded. The format is a list of objects including name, displayname and matchers (containing keys corresponding value)."
```
Comment (ArtieReus) [inline alerts/plugindefinition.yaml:41]: Use please same description as in the readme

File: alerts/plugindefinition.yaml [hunk 4]
```diff
@@ -37,6 +37,10 @@ spec:
...
+      required: false
```
Comment (ArtieReus) [inline alerts/plugindefinition.yaml:42]: You should be aware to find a solution in the UI if no PredefinedFilters are set...

**Review by ArtieReus — CHANGES_REQUESTED** [review]

File: alerts/plugindefinition.yaml (hunk 4 above)
Comment (TilmanHaupt) [reply alerts/plugindefinition.yaml:42]: not in this PR but yes.

File: alerts/plugindefinition.yaml [hunk 5]
```diff
@@ -34,7 +34,11 @@ spec:
       required: false
       type: list
     - name: silenceTemplates
-      description: "SilenceTemplates are pre-defined silence templates that can be used to scheduled Maintenance Windows. The format consists of a list of objects including description, editable_labels (array of strings specifying the labels that users can modify), fixed_labels (map containing fixed labels and their corresponding values), status, and title."
+      description: "SilenceTemplates are pre-defined silence templates that can be used to scheduled maintenance windows. The format consists of a list of objects including description, editable_labels (array of strings specifying the labels that users can modify), fixed_labels (map containing fixed labels and their corresponding values), status, and title."
+      required: false
+      type: list
+    - name: predefinedFilters
+      description: "PredefinedFilters are filters applied through TabNavigation in the UI to differentiate between contexts such as Production, QA, and others. They are loaded by default when the application is loaded. The format is a list of objects including name, displayname and matchers (containing keys corresponding value)"
```
Comment (IvoGoman) [inline alerts/plugindefinition.yaml:41]:
```suggestion
      description: "PredefinedFilters are filters applied through TabNavigation in the UI to differentiate between contexts such as Production, QA, and others. They are loaded by default when the application is loaded. The format is a list of objects including name, display name and matchers (map containing matchers as name and expression pairs)"
```

File: alerts/README.md [hunk 6]
```diff
@@ -43,6 +43,10 @@ This plugin extension contains [Prometheus Alertmanager](https://github.com/prom
...
 | `filterLabels`          | FilterLabels are the labels shown in the filter dropdown, enabling users to filter alerts based on specific criteria. The 'Status' label serves as a default filter, automatically computed from the alert status attribute and will be not overwritten. The labels must be an array of strings. Example: `["app", "cluster", "cluster_type"]` | `[]`           |
 
+| `silenceTemplates` | SilenceTemplates are used in the Modal (schedule silence) to allow pre-defined silences to be used to scheduled maintenance windows. The format consists of a list of objects including description, editable_labels (array of strings specifying the labels that users can modify), fixed_labels (map containing fixed labels and their corresponding values), status, and title. | `[]` |
+
+| `predefinedFilters` | PredefinedFilters are filters applied through TabNavigation in the UI to differentiate between contexts such as Production, QA, and others. They are loaded by default when the application is loaded. The format is a list of objects including name, displayname and matchers (containing keys corresponding value) | `[]` |
```
Comment (IvoGoman) [inline alerts/README.md:48]:
```suggestion
| `predefinedFilters` | PredefinedFilters are filters applied through TabNavigation in the UI to differentiate between contexts such as Production, QA, and others. They are loaded by default when the application is loaded. The format is a list of objects including name, displayname and matchers (map containing matchers as name and expression pairs) | `[]` |
```

**Review by IvoGoman — APPROVED** [review]

**Pull request merged**